src/ai-sdk-provider.ts

```diff
@@ -21,9 +21,15 @@ export async function createTinfoilAI(apiKey: string, options: CreateTinfoilAIOp
 
   await secureClient.ready();
 
+  // Get the baseURL from SecureClient after initialization
+  const finalBaseURL = baseURL || secureClient.getBaseURL();
+  if (!finalBaseURL) {
+    throw new Error("Unable to determine baseURL for AI SDK provider");
+  }
+
   return createOpenAICompatible({
     name: "tinfoil",
-    baseURL: baseURL.replace(/\/$/, ""),
+    baseURL: finalBaseURL.replace(/\/$/, ""),
```

Comment:
do we need to be replacing "/" if it's already processed by secureClient? overall it feels a bit strange to me to first pass the url through secureClient and then process it down the line agaiin

Reply:
Good point, we should probably refactor this so all the URL parsing is done a little differently.

Do we still need a separate base URL?